File: backend/pkg/model/discipline.go
package model

type Discipline struct {
	Id   int    `json:"discipline_id" db:"id"`
	Name string `json:"name" db:"name"`
}

type DisciplineId struct {
	Id int `json:"discipline_id" binding:"required"`
}

type AddNewDiscipline struct {
	Name                string `json:"discipline_name" binding:"required"`
	NameEn              string `json:"discipline_name_en" binding:"required"`
	SeminarVisitingMark int    `json:"seminar_visiting_mark"`
	LessonVisitingMark  int    `json:"lesson_visiting_mark"`
	ExamMark            int    `json:"exam_mark"`
}

type DisciplineInfo struct {
	Id                  int    `json:"id" db:"id"`
	Name                string `json:"discipline_name" db:"name"`
	NameEn              string `json:"discipline_name_en" db:"name_en"`
	SeminarVisitingMark *int   `json:"seminar_visiting_mark" db:"seminar_visiting_mark"`
	LessonVisitingMark  *int   `json:"lesson_visiting_mark" db:"lesson_visiting_mark"`
	ExamMark            int    `json:"exam_mark" db:"exam_mark"`
}

type DisciplineInfoDoubleLang struct {
	Id                  int    `json:"id" db:"id"`
	Name                string `json:"discipline_name" db:"name"`
	SeminarVisitingMark *int   `json:"seminar_visiting_mark" db:"seminar_visiting_mark"`
	LessonVisitingMark  *int   `json:"lesson_visiting_mark" db:"lesson_visiting_mark"`
	ExamMark            int    `json:"exam_mark" db:"exam_mark"`
}

type InputGroupToDiscipline struct {
	GroupId      int `json:"group_id" binding:"required"`
	DisciplineId int `json:"discipline_id" binding:"required"`
}

type DisciplineForGroup struct {
	Id   int    `json:"discipline_id" db:"id"`
	Name string `json:"name" db:"name"`
}

type DisciplinesResponseRu struct {
}

type DisciplinesResponse struct {
	Ru []Discipline `json:"ru"`
	En []Discipline `json:"en"`
}

type DisciplineInput struct {
	Id int `json:"discipline_id" binding:"required"`
}

type DisciplineNewNameInput struct {
	Id   int    `json:"discipline_id" binding:"required"`
	Name string `json:"name" binding:"required"`
}

File: frontend/src/models/discipline/discipline.ts
import { II18n } from 'src/services/resource/i18nTypes';

export type IDiscipline = IDiscipline.Discipline;

export namespace IDiscipline {
  export interface Discipline {
    name: string;
    discipline_id: string;
  }

  export interface DisciplineFullInfo {
    discipline_name: string;
    discipline_name_en: string;
    exam_mark: number;
    id: number;
    lesson_visiting_mark: number;
    seminar_visiting_mark: number;
  }

  export type DisciplineFetchData = II18n<IDiscipline[]>;

  export interface FetchData {
    id: string;
  }

  export interface AddGroup {
    discipline_id: number;
    group_id: number;
  }

  export interface GetGroups {
    id: number;
  }

  export interface CreateDiscipline {
    discipline_name: string;
    discipline_name_en: string;
    exam_mark: number;
    lesson_visiting_mark: number;
    seminar_visiting_mark: number;
  }

  export interface ChangeDisciplineName {
    discipline_id: number;
    name: string;
    name_en: string;
  }

  export interface ChangeDisciplineExamMark {
    discipline_id: number;
    exam_mark: number;
  }

  export interface ChangeDisciplineLessonMark {
    discipline_id: number;
    lesson_mark: number;
  }

  export interface ChangeDisciplineSeminarMark {
    discipline_id: number;
    seminar_mark: number;
  }
}
